internal/service/flexcluster/resource.go

```diff
@@ -196,3 +205,27 @@ func splitFlexClusterImportID(id string) (projectID, clusterName *string, err er
 
 	return
 }
+
+func isUpdateAllowed(ctx context.Context, plan, state *TFModel) (bool, error) {
+	if !plan.ProjectId.Equal(state.ProjectId) {
+		return false, errors.New("project_id cannot be updated")
+	}
+	if !plan.Name.Equal(state.Name) {
+		return false, errors.New("name cannot be updated")
+	}
+	planProviderSettings := &TFProviderSettings{}
+	if diags := plan.ProviderSettings.As(context.Background(), planProviderSettings, basetypes.ObjectAsOptions{}); diags.HasError() {
+		return false, errors.New("provider_settings cannot be updated")
+	}
+	stateProviderSettings := &TFProviderSettings{}
+	if diags := state.ProviderSettings.As(ctx, stateProviderSettings, basetypes.ObjectAsOptions{}); diags.HasError() {
+		return false, errors.New("provider_settings cannot be updated")
+	}
+	if !planProviderSettings.BackingProviderName.Equal(stateProviderSettings.BackingProviderName) {
+		return false, errors.New("backing_provider_name cannot be updated")
+	}
+	if !planProviderSettings.RegionName.Equal(stateProviderSettings.RegionName) {
+		return false, errors.New("region_name cannot be updated")
```

Comment:
q: Was surprised to see that region_name cannot be updated, was curious if we tested the API just to confirm this. (cc @cveticm as we have spoke about this)

Reply:
I tested it, region_name cannot be updated and was also confirmed by the upstream team

Reply:
thanks for confirming 👍